server/asset/eth/rpcclient.go

```diff
@@ -171,11 +197,18 @@ func (c *rpcclient) withTokener(assetID uint32, f func(*tokener) error) error {
 // bestHeader gets the best header at the time of calling.
 func (c *rpcclient) bestHeader(ctx context.Context) (hdr *types.Header, err error) {
```

Comment:
To clarify discussion on matrix happening now, I think `poll` will work best without calling `bestHeader`:

```diff
diff --git a/server/asset/eth/eth.go b/server/asset/eth/eth.go
index 38dbfe683..34fa87db3 100644
--- a/server/asset/eth/eth.go
+++ b/server/asset/eth/eth.go
@@ -708,13 +708,19 @@ func (eth *ETHBackend) poll(ctx context.Context) {
                        be.sendBlockUpdate(u)
                }
        }
-       bhdr, err := eth.node.bestHeader(ctx)
+       blockNum, err := eth.node.blockNumber(ctx)
        if err != nil {
-               send(false, fmt.Errorf("error getting best block header from geth: %w", err))
+               send(false, fmt.Errorf("error getting best block number: %w", err))
+               return
+       }
+       if blockNum != best.height {
+               // Same height, nothing to do.
                return
        }
-       if bhdr.Hash() == best.hash {
-               // Same hash, nothing to do.
+
+       bhdr, err := eth.node.bestHeader(ctx)
+       if err != nil {
+               send(false, fmt.Errorf("error getting best block header from geth: %w", err))
                return
        }
        update := func(reorg bool, fastBlocks bool) {
```

That should drastically reduce RPC usage because `eth_blockNumber` is very cheap, but `eth_getBlockByNumber` is relatively costly in terms of bandwidth.

Reply:
Oh, whoops I meant `==`, not `!=`